### PR TITLE
fix scatter plot markers ACID(83)

### PR DESCRIPTION
**Pull request by PeterPablo**
Before: `scatterPlotMarkers` (`ACID(83)`) would cause failed plot in octave.
Now: Plot works, though `matlab2tikz` conversion is still not correct.

The markers will be present after `octaveGuessType()` #669 is merged.

### CODE REVIEW

**Comment by PeterPablo** [comment]
MATLAB hashes are unchanged (checked HG1 and HG2).
I would merge this and research why the octave output is an empty axis later. Goal of this PR is to fix the plot error.

**Comment by okomarov** [comment]
Haven't followed the whole thing, just make sure that the hash still fails to keep track that empty plot needs to be fixed :)

Sent from my iPhone

On 11 May 2015, at 15:51, "Peter Pablo" <notifications@github.com<mailto:notifications@github.com>> wrote:


MATLAB hashes are unchanged (checked HG1 and HG2).
I would merge this and research why the octave output is an empty axis later. Goal of this PR is to fix the plot error.

**Comment by PeterPablo** [comment]
I already added a `TODO` note. This PR is intended as a step towards fixed ACID tests. Looking at the graphical output and the code notes describe the current situation. I prefer to not mark the test as failing by purposely entering a wrong hash.

**Comment by PeterPablo** [comment]
image
This is the current situation. Previously the plot wouldn't even render in octave.

**Comment by egeerardyn** [comment]
@PeterPablo : please make sure that you are compiling the ACID suite with `pgfplots 1.12.1` to avoid problems with LuaLaTeX.

@okomarov We shouldn't mess with the hashes to indicate a status of the plot. If it is bad behavior, please add `stat.unreliable = true` and a TODO with a description of the problem. You can also add such information to the PDF report if you want.

**Comment by okomarov** [comment]
@egeerardyn I totally agree. 

:+1:

**Comment by PeterPablo** [comment]
I updated the initial post to reflect the current status. Is there something that needs to be done from my side?

File: test/suites/ACID.m [hunk 1]
```diff
@@ -1176,11 +1176,12 @@
 % =========================================================================
 function [stat] = scatterPlotMarkers()
   stat.description = 'Scatter plot with with different marker sizes and legend.';
-  stat.unreliable = isOctave;
+  stat.unreliable = isOctave; % FIXME: Output is empty?!
```
Comment (PeterPablo) [inline test/suites/ACID.m]: Will be fixed by #669.

File: test/suites/ACID.m [hunk 2]
```diff
@@ -1176,11 +1176,12 @@
...
 
   n = 1:10;
   d = 10;
   s = d^2 * n;
+  if isOctave(), s = s/25; end; % smaller size in octave
```
Comment (egeerardyn) [inline test/suites/ACID.m]: I think this deserves a more elaborate explanation...

**Comment by PeterPablo** [comment]
image

As you see I added comments as to why a scaling of the scatter size is needed. MATLAB seems to expect an area (unit points squared) and octave a radius (unit points), thus I went for taking the square root-
Strictly the square root is only correct for squares... But let's go with this, since I don't know how MATLAB/octave handle this internally.

Please let's get this in before #669, as `ACID(83)` currently throws an error in octave and will be affected by the new approach of #669.

@egeerardyn, please review. MATLAB behavior is unchanged, so octave only.

**Comment by PeterPablo** [comment]
Thanks for taking a look :-) The pile of PR is getting smaller...

File: test/suites/ACID.m [hunk 3]
```diff
@@ -1195,11 +1195,18 @@
...
 
   n = 1:10;
   d = 10;
-  s = d^2 * n;
+  s = d^2 * n; % scatter size
+  % MATLAB: Use the default area of 36 points squared. The units for the
+  %         marker area is points squared.
+  % octave: If s is not given, [...] a default value of 8 points is used.
+  % Try obtain similar behavior and thus apply square root: sqrt(36) vs. 8
+  s_orig = s;
```
Comment (egeerardyn) [inline test/suites/ACID.m]: Perhaps we should come up with a better name than `s_orig`? E.g.  `sArea` vs `sDiameter` to indicate the dimensions of the variable?

**Comment by egeerardyn** [comment]
# R2014a
Again `ACID(88)`:            ACID( 88) annotationSubplots       : hash 26ec9a1ad89e3e9438401491e51961e2 != (11bba1b00e8aa4b3e0643e52a79fb1dd)

```matlab
% This file was created by matlab2tikz.
%
\documentclass[tikz]{standalone}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{pgfplots}
\usepackage{grffile}
\pgfplotsset{compat=newest}
\usetikzlibrary{plotmarks}
\usetikzlibrary{arrows.meta}
\usepgfplotslibrary{patchplots}
\usepackage{amsmath}

\newlength\figureHeight \setlength{\figureHeight}{6cm}
\newlength\figureWidth \setlength{\figureWidth}{10cm}
\begin{document}
\definecolor{mycolor1}{rgb}{0.00000,0.44700,0.74100}%
%
\begin{tikzpicture}

\begin{axis}[%
width=0.282\figureWidth,
height=0.583\figureHeight,
at={(0\figureWidth,0.376\figureHeight)},
scale only axis,
xmin=   0,
xmax=   1,
ymin=   0,
ymax=   1,
title={$\text{f(x)=x}^\text{2}$}
]
\addplot [color=mycolor1,solid,forget plot]
  table[row sep=crcr]{%
   0	   0\\
0.03	0.0009\\
0.06	0.0036\\
0.09	0.0081\\
0.12	0.0144\\
0.15	0.0225\\
0.18	0.0324\\
0.21	0.0441\\
0.24	0.0576\\
0.27	0.0729\\
 0.3	0.09\\
0.33	0.1089\\
0.37	0.1369\\
0.41	0.1681\\
0.45	0.2025\\
0.49	0.2401\\
0.53	0.2809\\
0.57	0.3249\\
0.61	0.3721\\
0.65	0.4225\\
0.69	0.4761\\
0.73	0.5329\\
0.77	0.5929\\
0.81	0.6561\\
0.85	0.7225\\
0.89	0.7921\\
0.94	0.8836\\
0.99	0.9801\\
   1	   1\\
};
\end{axis}

\begin{axis}[%
width=0.243\figureWidth,
height=0.583\figureHeight,
at={(0.422\figureWidth,0\figureHeight)},
scale only axis,
xmin=   0,
xmax=   1,
ymin=   0,
ymax=   1
]
\addplot [color=mycolor1,solid,forget plot]
  table[row sep=crcr]{%
   0	   0\\
0.17	0.00083521\\
0.24	0.00331776\\
0.29	0.00707281\\
0.33	0.0118592\\
0.37	0.0187416\\
 0.4	0.0256\\
0.43	0.034188\\
0.46	0.0447746\\
0.49	0.057648\\
0.52	0.0731162\\
0.55	0.0915063\\
0.58	0.113165\\
0.61	0.138458\\
0.63	0.15753\\
0.65	0.178506\\
0.67	0.201511\\
0.69	0.226671\\
0.71	0.254117\\
0.73	0.283982\\
0.75	0.316406\\
0.77	0.35153\\
0.79	0.389501\\
0.81	0.430467\\
0.83	0.474583\\
0.85	0.522006\\
0.88	0.599695\\
0.91	0.68575\\
0.94	0.780749\\
0.97	0.885293\\
   1	   1\\
};
\end{axis}

\begin{axis}[%
width=0.238\figureWidth,
height=0.621\figureHeight,
at={(0.762\figureWidth,0.282\figureHeight)},
scale only axis,
xmin=   0,
xmax=   1,
ymin=   0,
ymax=   1
]
\addplot [color=mycolor1,solid,forget plot]
  table[row sep=crcr]{%
   0	   0\\
0.01	0.316228\\
0.02	0.37606\\
0.03	0.416179\\
0.04	0.447214\\
0.05	0.472871\\
0.06	0.494923\\
0.08	0.53183\\
 0.1	0.562341\\
0.12	0.588566\\
0.14	0.611691\\
0.16	0.632456\\
0.19	0.66022\\
0.22	0.684866\\
0.25	0.707107\\
0.29	0.733837\\
0.33	0.757929\\
0.37	0.779921\\
0.42	0.80503\\
0.47	0.827989\\
0.53	0.853236\\
0.59	0.876421\\
0.66	0.901334\\
0.74	0.927487\\
0.82	0.951598\\
0.91	0.976698\\
   1	   1\\
};
\end{axis}

\begin{axis}[%
width=1.134\figureWidth,
height=1.134\figureHeight,
at={(-0.079\figureWidth,-0.079\figureHeight)},
scale only axis,
xmin=   0,
xmax=   1,
ymin=   0,
ymax=   1,
hide axis,
axis x line*=bottom,
axis y line*=left
]
\node[below right, align=left, text=black, draw=black]
at (rel axis cs:0.367,0.552) {$\text{f}^\text{2}$};
\addplot [color=white,solid,forget plot]
  table[row sep=crcr]{%
0.3263	0.6606\\
0.424425	0.363045\\
};

\addplot[area legend,solid,table/row sep=crcr,patch,fill=white,faceted color=black,forget plot,patch table={%
   0	   1	   2\\
   3	   4	   5\\
}]
table[row sep=crcr] {%
x	y\\
0.425018	0.374347\\
0.4281	0.3519\\
0.424425	0.363045\\
0.4281	0.3519\\
0.419874	0.363745\\
0.424425	0.363045\\
};
\addplot [color=white,solid,forget plot]
  table[row sep=crcr]{%
0.6766	0.3108\\
0.720948	0.619705\\
};

\addplot[area legend,solid,table/row sep=crcr,patch,fill=white,faceted color=black,forget plot,patch table={%
   0	   1	   2\\
   3	   4	   5\\
}]
table[row sep=crcr] {%
x	y\\
0.71676	0.615199\\
0.7229	0.6333\\
0.720948	0.619705\\
0.7229	0.6333\\
0.723035	0.609569\\
0.720948	0.619705\\
};
\node[below, align=center, text=white]
at (axis cs:0.677,0.311) {invert};
\end{axis}
\end{tikzpicture}%
\end{document}
```

# R2014b on mac
Perfect score!

# Octave 3.8.0 Mac
Perfect score!

**Comment by PeterPablo** [comment]
Please rerun to verify the hashes. This was most likely the same issue as in the other PR (outdated hashes). I'll come up with better naming inside the function.

**Comment by egeerardyn** [comment]
I guessed so. The hash is fine now (didn't check the rest, but those should be alright). Feel free to merge after you've changed the name. :+1:

**Pull request merged**